packages/node_modules/@webex/internal-plugin-services/src/service-catalog.js

```diff
@@ -0,0 +1,230 @@
+import AmpState from 'ampersand-state';
+
+import ServiceUrl from './service-url';
+
+/* eslint-disable no-underscore-dangle */
+/**
+ * @class
+ */
+const ServiceCatalog = AmpState.extend({
+  namespace: 'ServiceCatalog',
+
+  props: {
+    serviceGroups: ['object', true, (() => ({
+      discovery: [
+        new ServiceUrl({
+          name: 'u2c',
+          defaultUrl: process.env.U2C_SERVICE_URL ||
+            'https://u2c.wbx2.com/u2c/api/v1',
+          hosts: []
+        })
+      ],
+      preauth: [],
+      postauth: []
+    }))]
+  },
+
+  /**
+   * **Internal**
+   * Search the service url array to locate a `ServiceUrl`
+   * class object based on its name.
+   * @param {string} name
+   * @param {string} [serviceGroup]
+   * @returns {ServiceUrl}
+   */
+  _getUrl(name, serviceGroup) {
+    const serviceUrls = (typeof serviceGroup === 'string') ?
+      this.serviceGroups[serviceGroup] || [] :
+      [
+        ...this.serviceGroups.discovery,
+        ...this.serviceGroups.preauth,
+        ...this.serviceGroups.postauth
+      ];
+
+    return serviceUrls.find((serviceUrl) => serviceUrl.name === name);
+  },
+
+  /**
+   * **Internal**
+   * Safely load one or more `ServiceUrl`s into this `Services` instance.
+   * @param {string} serviceGroup
+   * @param  {Array<ServiceUrl>} services
+   * @returns {Services}
+   */
+  _loadServiceUrls(serviceGroup, services) {
+    // declare namespaces outside of loop
+    let existingService;
+
+    services.forEach((service) => {
+      existingService = this._getUrl(service.name, serviceGroup);
+
+      if (!existingService) {
+        this.serviceGroups[serviceGroup].push(service);
+      }
+    });
+
+    return this;
+  },
+
+  /**
+   * **Internal**
+   * Safely unload one or more `ServiceUrl`s into this `Services` instance
+   * @param {string} serviceGroup
+   * @param  {Array<ServiceUrl>} services
+   * @returns {Services}
+   */
+  _unloadServiceUrls(serviceGroup, services) {
+    // declare namespaces outside of loop
+    let existingService;
+
+    services.forEach((service) => {
+      existingService = this._getUrl(service.name, serviceGroup);
+
+      if (existingService) {
+        this.serviceGroups[serviceGroup].splice(
+          this.serviceGroups[serviceGroup].indexOf(existingService), 1
+        );
+      }
+    });
+
+    return this;
+  },
+
+  /**
+   * Get a service url from the current services list by name.
+   * @param {string} name
+   * @param {boolean} priorityHost
+   * @param {string} serviceGroup
+   * @returns {string}
+   */
+  get(name, priorityHost, serviceGroup) {
```

Comment:
Can we add a 'cluster' parameter here that defaults to the home cluster?

Reply:
@mwegman, `serviceGroup` should route to different catalogs from each URL, `/limited`, `/signin`, and `/`, as well as the single discovery URL for **U2C**. Currently, we don't receive specific cluster information from the catalog, but, if you'd like to grab host information, this can be done multiple ways:

`services.get('serviceName', true);` will return the highest priority host URL
`services.get('serviceName');` will return the default URL for the service.

Below is a reference, this is what data we must use to set up the services catalog:

```
{
  "serviceLinks": {
    "metrics": "https://metrics-a.wbx2.com/metrics/api/v1" // used in url when !priorityHost
  },
  "hostCatalog": {
    "metrics-a.wbx2.com": [
      {
        "host": "metrics.a6.ciscospark.com", // used in url when priorityHost === true
        "ttl": -1,
        "priority": 6
      },
      {
        "host": "metrics.a7.ciscospark.com",
        "ttl": -1,
        "priority": 5
      }
    ]
  },
  "format": "hostmap"
}
```

I will reach out to you via Webex Teams.